src/render/tutorials/api-gateway-security/2_0.html.md

```diff
@@ -0,0 +1,913 @@
+
+```
+$ gradlew clean build-docker
+```
+
+and run it:
+
+```
+$ docker run -p8092:8092 knotx-example/secure-api-gateway
+```
+
+After a while the Docker container should be up and running.
+
+Let's try accessing our endpoints:
+
+```
+$ curl -X GET http://localhost:8092/api/secure/basic
+Unauthorized
+```
+
+As expected - we were not authorized to access this endpoint.
+
+Let's try again with a valid `Authorization` header. The header should be a Base64-encoded string of pair of `username:password` prepended with `Basic` keyword,  as described [here](https://docs.apigee.com/api-platform/system-administration/basic-auth#basic-authentication-format):
+
+```
+$ curl -X GET http://localhost:8092/api/secure/basic -H 'Authorization: Basic am9objpzM2NyM3Q='
+{"message":"Hello World From Knot.x with Basic Auth!"}
+```
+
+### Functional tests
+
+It's always a good practice to test our code. Let's implement some tests that will automatically do the same things we did manually above.
+
+The tests we're about to write will be functional tests, that will be executed on a real, running Docker image, just after the `healthcheck` (described in [this tutorial](http://knotx.io/tutorials/getting-started-with-docker/2_0/)) executes successfully.
+
+Let's add a new file: `functional/src/test/java/com/project/test/functional/BasicAuthITCase.java`
+
+```java
+class BasicAuthITCase {
+
+  @Test
+  @DisplayName("GIVEN no authorization WHEN call basicAuth API EXPECT Unauthorized")
+  void givenNoAuthorizationWhenCallBasicAuthApiExpectUnauthorized() {
+    given()
+        .port(8092)
+      .when()
+        .get("/api/secure/basic")
+      .then()
+        .assertThat()
+        .statusCode(401);
+  }
+
+  @Test
+  @DisplayName("GIVEN authorization WHEN call basicAuth API EXPECT Ok")
+  void givenAuthorizationWhenCallBasicAuthApiExpectOk() {
+    given()
+        .port(8092)
+        .header("Authorization", "Basic am9objpzM2NyM3Q=")
+      .when()
+        .get("/api/secure/basic")
+      .then()
+        .assertThat()
+        .statusCode(200);
+  }
+}
+```
+
+We've created two simple tests using [REST-assured](http://rest-assured.io/). They do the same actions we did above:
+
+- call secured endpoint without authorization and expect `401 - Unauthorized` response
+- call the same endpoint with authorization and expect `200 - OK` response
+
+If you try rebuilding the project now (remember to stop any running containers before that) you'll notice that those test will be invoked. Now we're confident that if our future changes will somehow disable or modify our `/api/secure/basic` logic, we'll know at build time!
+
+## JWT
+
+Now let's do something fancier than just passing the credentials. We'll use [JSON Web Tokens](https://jwt.io/).
+
+A JWT (JSON Web Token) is a well-known solution for API authorization.
+The JWT allows verifying the owner of some JSON data. The JSON data stored in a JWT is serialized, not encrypted. So it can be seen by anyone so always use HTTPS.
```

Comment:
```suggestion
The JWT allows verifying the owner of some JSON data. The JSON data stored in a JWT is serialized, not encrypted. So it can be seen by anyone (remember to always use HTTPS).
```